src/hooks/usePastTransaction/usePastTransaction.tsx

```diff
@@ -3,40 +3,49 @@ import { PastTransactionsResult } from "../../types";
 import { usePrevious } from "../usePrevious";
 import { getPastTransactions } from "../../services/quota";
 import { Sentry } from "../../utils/errorTracking";
+import { ERROR_MESSAGE } from "../../context/alert";
 
 export type PastTransactionHook = {
-  pastTransactionsResult: PastTransactionsResult | null;
+  pastTransactionsResult: PastTransactionsResult["pastTransactions"] | null;
+  error: { message: string } | null;
 };
 
 export const usePastTransaction = (
-  id: string,
+  ids: string[],
   authKey: string,
   endpoint: string
 ): PastTransactionHook => {
-  const [
-    pastTransactionsResult,
-    setPastTransactionsResult
-  ] = useState<PastTransactionsResult | null>(null);
-  const prevId = usePrevious(id);
+  const [pastTransactionsResult, setPastTransactionsResult] = useState<
+    PastTransactionsResult["pastTransactions"] | null
+  >(null);
+  const [error, setError] = useState<{ message: string } | null>(null);
+  const prevIds = usePrevious(ids);
 
   useEffect(() => {
     const fetchPastTransactions = async (): Promise<void> => {
-      const pastTransactionsResponse = await getPastTransactions(
-        id,
-        authKey,
-        endpoint
-      );
-
-      setPastTransactionsResult(pastTransactionsResponse);
+      try {
+        const pastTransactionsResponse = await getPastTransactions(
+          ids,
+          authKey,
+          endpoint
+        );
+        setPastTransactionsResult(pastTransactionsResponse?.pastTransactions);
+      } catch (error) {
+        Sentry.captureException(`Unable to fetch past transactions: ${ids}`);
+        setError(new Error(ERROR_MESSAGE.PAST_TRANSACTIONS_ERROR));
+      }
     };
 
-    if (prevId != id)
-      fetchPastTransactions().catch(() => {
-        Sentry.captureException(`Unable to fetch past transactions: ${id}`);
-      });
-  }, [authKey, endpoint, id, prevId]);
+    if (
+      typeof prevIds !== typeof ids ||
+      (prevIds && ids && !prevIds.every(id => ids.includes(id)))
```

Comment:
Hmm does `prevIds !== ids` work?
Also, the 2nd condition implies that if `ids` has an additional id appended to `prevIds`, then `fetchPastTransactions` won't be called, when we want it to be called

Reply:
No eh, cuz `[1] !== [1]`. We were doing the fetch multiple times with just `prevIds !== ids`.

You're right, let me do a equal length check too

Reply:
In this case the passed in `ids` won't change reference since it comes from the ids state in `CustomerQuotaScreen`. So when the effect runs, the fetch won't be called unless a new id was added.

I think checking each value is useful, but doesn't work too well if we're comparing arrays of objects. I'm wondering what's a good pattern/guideline for these sort of checks as we use this elsewhere too (e.g. useCart).

Reply:
Hmm that's weird but i logged it and observed that even though the ids stayed the same, the fetch kept happening. I am not so sure about the data path here but let me test it again.

Reply:
Turns out it was the unit test that was set up wrongly. Original `prevIds !== ids` check would have sufficed for the app itself. Reverted :)

Reply:
Oh saw the offending test, a good reminder too haha